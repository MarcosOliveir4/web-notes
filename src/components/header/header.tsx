import { Divider, Flex, Heading, Spacer } from "@chakra-ui/react";

import { ToggleTheme } from "../toggleTheme";

interface HeaderProps {
  title: string;
}

export const Header = ({ title }: HeaderProps) => {
  return (
    <>
      <Flex p={5} data-testid="header">
        <Heading data-testid="header-title">{title}</Heading>
        <Spacer />
        <ToggleTheme />
      </Flex>
      <Divider />
    </>
  );
};
