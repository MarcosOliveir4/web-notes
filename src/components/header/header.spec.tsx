import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { ThemeWrapper } from "../../tests/utils";
import { Header } from "./header";

const titleDefault = "web notes";
const sut = (title = titleDefault) =>
  render(<Header title={title} />, { wrapper: ThemeWrapper });

describe("<Header />", () => {
  it("should match snapshot", () => {
    sut();
    const header = screen.getByTestId("header");
    expect(header).toMatchSnapshot();
  });
  it("should render title", () => {
    sut();
    const title = screen.getByTestId("header-title");
    expect(title).toHaveTextContent(titleDefault);
  });
  it("should render toggle theme", () => {
    sut();
    const toggleTheme = screen.getByTestId("toggleTheme");
    expect(toggleTheme).toBeTruthy();
  });
});
