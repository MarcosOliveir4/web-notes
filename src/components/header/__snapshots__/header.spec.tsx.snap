// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header /> > should match snapshot 1`] = `
<div
  class="css-1k27jlm"
  data-testid="header"
>
  <h2
    class="chakra-heading css-1dklj6k"
    data-testid="header-title"
  >
    web notes
  </h2>
  <div
    class="css-17xejub"
  />
  <button
    class="chakra-button css-ez23ye"
    data-testid="toggleTheme"
    type="button"
  >
    <svg
      data-testid="moonIcon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
      />
    </svg>
  </button>
</div>
`;
